mrp/src/mrp/__init__.py

```diff
@@ -24,17 +25,36 @@ def main(*args):
     sys.exit(0)
 
 
-def import_msetup(path):
-    # TODO(lshamis): Maybe add args to filter imported processes.
+def import_msetup(path, processes=None):
+    # Get path relative to caller.
     caller_path = inspect.stack()[1].filename
     caller_dir = os.path.dirname(caller_path)
     target_path = os.path.join(caller_dir, path)
 
+    # Add default filename if provided path is a directory.
     if os.path.isdir(target_path):
         target_path = os.path.join(target_path, "msetup.py")
 
+    # We hook mrp.process to filter processes within the import.
+
+    # Save the real function for use within the hook
+    # and to restore post-import.
+    process_impl = mrp.process
+
+    # Replacement implementation.
+    def hook(name, *args, **kwargs):
+        if processes is None or name in processes:
+            return process_impl(name, *args, **kwargs)
+
+    # Attach the hook.
+    mrp.process = hook
+
+    # Import.
     SourceFileLoader("msetup", target_path).load_module()
 
+    # Undo the hook.
+    mrp.process = process_impl
+
```

Comment:
1. Curious if there's a reason to go through the relatively opaque method of hooking `mrp.process`. Why not simply do something like `loader = SourceFileLoader("msetup", target_path); [loader.load_module(name) for name in processes]`?
2. I noticed [`SourceFileLoader.load_module`](https://docs.python.org/3/library/importlib.html#importlib.machinery.SourceFileLoader.load_module) is being deprecated, is that something of concern?

Reply:
I don't think the string passed to `load_module` is the name of the symbol to import. Even if it was, processes aren't Python symbols. They are entries in the `defined_processes` table.